src/api/gists.rs

```diff
@@ -79,6 +79,23 @@ impl<'octo> GistsHandler<'octo> {
         self.crab.get(format!("/gists/{id}"), None::<&()>).await
     }
 
+    /// Delete a single gist.
+    ///
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// octocrab::instance().gists().delete("00000000000000000000000000000000").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn delete(&self, gist_id: impl AsRef<str>) -> Result<bool> {
+        let gist_id = gist_id.as_ref();
+        let response = self
+            .crab
+            ._delete(format!("/gists/{gist_id}"), None::<&()>)
+            .await?;
+        Ok(response.status() == http::StatusCode::NO_CONTENT)
```

Comment:
https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#delete-a-gist--status-codes

funnily enough, it returns a 404 on error, does that get transformed into the `Error(...)` variant here? If so the correct return type might be `Result<()>` since that is isomorphic to `bool`.

Noticed the same _issue_ with status codes from `GET /gists/{gist_id}/star`. WDYT? @XAMPPRocky

Reply:
Yes, `Result<()>` should be the type here, I think that makes the most sense.